code/modules/multiz/pipes.dm

```diff
@@ -0,0 +1,328 @@
+////////////////////////////
+// parent class for pipes //
+////////////////////////////
+obj/machinery/atmospherics/pipe/zpipe
+		icon = 'icons/obj/structures.dmi'
+		icon_state = "up"
+
+		name = "upwards pipe"
+		desc = "A pipe segment to connect upwards."
+
+		volume = 70
+
+		dir = SOUTH
+		initialize_directions = SOUTH
+
+		// node1 is the connection on the same Z
+		// node2 is the connection on the other Z
+
+		var/minimum_temperature_difference = 300
+		var/thermal_conductivity = 0 //WALL_HEAT_TRANSFER_COEFFICIENT No
+
+		var/maximum_pressure = 70*ONE_ATMOSPHERE
+		var/fatigue_pressure = 55*ONE_ATMOSPHERE
+		alert_pressure = 55*ONE_ATMOSPHERE
+
+
+		level = 1
+
+obj/machinery/atmospherics/pipe/zpipe/New()
+	..()
+	switch(dir)
+		if(SOUTH)
+			initialize_directions = SOUTH
+		if(NORTH)
+			initialize_directions = NORTH
+		if(WEST)
+			initialize_directions = WEST
+		if(EAST)
+			initialize_directions = EAST
+		if(NORTHEAST)
+			initialize_directions = NORTH
+		if(NORTHWEST)
+			initialize_directions = WEST
+		if(SOUTHEAST)
+			initialize_directions = EAST
+		if(SOUTHWEST)
+			initialize_directions = SOUTH
+
+/obj/machinery/atmospherics/pipe/zpipe/hide(var/i)
+	if(istype(loc, /turf/simulated))
+		invisibility = i ? 101 : 0
+	update_icon()
+
+obj/machinery/atmospherics/pipe/zpipe/process()
+	if(!parent) //This should cut back on the overhead calling build_network thousands of times per cycle
+		..()
+	else
+		. = PROCESS_KILL
+
+obj/machinery/atmospherics/pipe/zpipe/check_pressure(pressure)
+	var/datum/gas_mixture/environment = loc.return_air()
+
+	var/pressure_difference = pressure - environment.return_pressure()
+
+	if(pressure_difference > maximum_pressure)
+		burst()
+
+	else if(pressure_difference > fatigue_pressure)
+		//TODO: leak to turf, doing pfshhhhh
+		if(prob(5))
+			burst()
+
+	else return 1
+
+obj/machinery/atmospherics/pipe/zpipe/proc/burst()
+	src.visible_message("<span class='warning'>\The [src] bursts!</span>");
+	playsound(src.loc, 'sound/effects/bang.ogg', 25, 1)
+	var/datum/effect/effect/system/smoke_spread/smoke = new
+	smoke.set_up(1,0, src.loc, 0)
+	smoke.start()
+	qdel(src) // NOT qdel.
```

Comment:
what did he mean by this

Reply:
I've got no clue. I saw that, but decided not to touchy.

Reply:
I'm going to go out on a limb here and guess someone badcoded this as `qdel.src` by accident and that comment was someone else being cheeky in a bugfix

Reply:
`qdel.src` wouldn't compile
They probably wrote `del(src)` and it got regexed to `qdel` and nobody noticed the comment